kore/src/Kore/Repl/Interpreter.hs

```diff
@@ -463,6 +463,40 @@ tryAxiomClaim eac = do
     singleton :: a -> [a]
     singleton a = [a]
 
+    showUnificationFailure
+        :: Either [Axiom level] [Claim level]
+        -> Graph.Node
+        -> ReplM level ()
+    showUnificationFailure ac node = do
+        let rw =
+                listToMaybe
+                    . fmap (left . getRewriteRule)
+                    . either (fmap unAxiom) (fmap unClaim) $ ac
+        case rw of
+            Nothing    -> putStrLn' "No axiom or claim found."
+            Just first -> do
+                Strategy.ExecutionGraph { graph } <- Lens.use lensGraph
+                let second = Graph.lab' . Graph.context graph $ node
+                strategyPattern
```

Comment:
As above, I'm not really sure what the purpose of this is, but it probably deserves to be extracted because it seems to be operating at a relatively low level.

Reply:
You are right, this function is becoming a mess. I will refactor some code here to clean it up.